syntax = "proto3";
package Protocol;

import "Enum.proto";
import "Struct.proto";

// CS - Client to Server
// SC - Server to Client
// BC - BroadCast to Client 
//		BC_GL_ - BroadCast_Global : 전체유저 call
//		BC_		- BroadCast		  : 특정 그룹에서 braodcast

//login
message CS_LOGIN
{
	bool bIsDevLogin = 1;
	string user_name = 2;
	string passwd = 3;
}

message SC_LOGIN
{
	bool success = 1;
	hyps_user_info user_info = 2;
	repeated hyps_pos_info players = 3;
}

// join
message CS_REGIST
{
	string user_name = 1;
	string passwd = 2;
}

message SC_REGIST
{
	bool success = 1;
	string user_name = 2;
}

// enter room
message CS_ENTER_ROOM
{
	string room_name = 1;
	uint64 userId = 2;
}

message SC_ENTER_ROOM
{
	bool success = 1;
	repeated hyps_user_info users = 2;
}

message BC_ENTER_ROOM_OTHERS
{
	bool success = 1;
	repeated hyps_user_info users = 2;
}

// enter game
message CS_ENTER_GAME
{
	uint64 player_id = 1;
}

message SC_ENTER_GAME
{
	bool success = 1;
	hyps_pos_info my_player = 2;
}

message CS_LEAVE_GAME
{

}

message SC_LEAVE_GAME
{

}

message BC_LEAVE_GAME
{

}

message SC_SPAWN
{
	repeated hyps_object_info players = 1;
}

message BC_SPAWN
{
	repeated hyps_object_info players = 1;
}

message SC_DESPAWN
{
	repeated uint64 object_ids = 1;
}

message BC_DESPAWN
{
	repeated uint64 object_ids = 1;
}

message CS_MOVE_OBJECT
{
	hyps_pos_info move_info = 1;
	hype_move_state move_state = 2;
}

message SC_MOVE_OBJECT
{
	hyps_pos_info move_info = 1;
	hype_move_state move_state = 2;
}

message CS_CHAT
{
	string msg = 1;
}

message SC_CHAT
{
	bool success = 1;
	uint64 userId = 2;
	string msg = 3;
}

message BC_GL_CHAT
{
	string msg = 1;
}

message CS_ECHO
{
	string msg = 1;
}

message BC_GL_ECHO
{
	string msg = 1;
}
